code/modules/shuttle/shuttle.dm

```diff
@@ -18,9 +18,18 @@
 	var/dwidth = 0	//position relative to covered area, perpendicular to dir
 	var/dheight = 0	//position relative to covered area, parallel to dir
 
+	var/i_know_what_im_doing = 0
+
 	//these objects are indestructable
 /obj/docking_port/Destroy()
-	return QDEL_HINT_LETMELIVE
+	// unless you assert that you know what you're doing. Horrible things
+	// may result.
+	if(!i_know_what_im_doing)
+		return QDEL_HINT_LETMELIVE
+	else
+		// If not removed immediately, it can interfere with the docking
+		// detection code, which is annoying and inconvinient.
+		return QDEL_HINT_HARDDEL_NOW
```

Comment:
The docking port was lingering long enough to interfere with later attempts to import new shuttles, because the docking code was picking up the existence of the doomed docking port and refusing to let the imported shuttle dock.